### PR TITLE
DCS-1459 fixing matching logic during confirmation

### PR DESCRIPTION
**Pull request by andrewrlee**
We should be reading prison number off first match rather than details from BASM. This allows us to process BASM arrivals that matched on PNC

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/prison/ConfirmArrivalDetail.kt [hunk 1]
```diff
@@ -23,6 +23,16 @@ data class ConfirmArrivalDetail(
...
 
+  @Schema(
+    description = "The offender's Prison number.",
+    example = "A1234AA",
+    pattern = "^[A-Za-z]\\d{4}[A-Za-z]{2}\$",
+    maxLength = 7
+  )
+  @field:Length(max = 7)
+  @field:Pattern(regexp = "^[A-Za-z]\\d{4}[A-Za-z]{2}\$", message = "Prison number is not valid")
+  val prisonNumber: String? = null,
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/prison/ConfirmArrivalDetail.kt:34]: Not actually used yet but is required to allow users to select which of a many matches refers to this arrival

File: src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/prison/temporaryabsences/TemporaryAbsenceService.kt [hunk 2]
```diff
@@ -35,11 +35,11 @@ class TemporaryAbsenceService(
   }
 
   fun confirmTemporaryAbsencesArrival(
-    offenderNo: String,
+    prisonNumber: String,
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/prison/temporaryabsences/TemporaryAbsenceService.kt:38]: correcting terminology

File: src/test/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/arrivals/ArrivalsServiceConfirmArrivalTest.kt [hunk 3]
```diff
@@ -43,50 +44,107 @@ class ArrivalsServiceConfirmArrivalTest {
...
+          status = "INACTIVE OUT",
+        )
+      )
+    )
     whenever(prisonService.admitOffenderOnNewBooking(any(), any())).thenReturn(INMATE_DETAIL_NO_UNIT)
     assertThatThrownBy {
       arrivalsService.confirmArrival(MOVE_ID, CONFIRMED_ARRIVAL_DETAIL_PROTOTYPE)
-    }.hasMessage("Prisoner: 'ABC123A' do not have assigned living unit")
+    }.hasMessage("Prisoner: '$PRISON_NUMBER' do not have assigned living unit")
   }
```
Comment (andrewrlee) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/arrivals/ArrivalsServiceConfirmArrivalTest.kt:64]: this test wasn't valid as it was relying on prison number provided from BASM when there was no match

File: src/test/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/arrivals/ArrivalsServiceConfirmArrivalTest.kt [hunk 4]
```diff
@@ -312,36 +380,27 @@ class ArrivalsServiceConfirmArrivalTest {
...
       isCurrentPrisoner = false,
-      potentialMatches = listOf(
-        PotentialMatch(
-          firstName = FIRST_NAME,
-          lastName = LAST_NAME,
-          dateOfBirth = DATE_OF_BIRTH,
-          prisonNumber = OFFENDER_NO,
-          pncNumber = null
-        )
-      )
```
Comment (andrewrlee) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/arrivals/ArrivalsServiceConfirmArrivalTest.kt:344]: this wasn't valid as this object was being returned from BASM with matches, when matches are only added later... maybe we should represent this as two different types, before and after being enriched?

**Review by louisenorris — APPROVED** [review]
LGTM

**Pull request merged**